### PR TITLE
refactor(networking/vip): replace openstack api with huaweicloud v1 api

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Replace the Openstack API with HuaweiCloud v1 API.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
reference: #2027 

**Special notes for your reviewer**:

**Release note**:

```release-note
1. replace openstack api with huaweicloud v1 api.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/vpc' TESTARGS='-run=TestAccNetworkingVip_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpc -v -run=TestAccNetworkingVip_ -timeout 360m -parallel 4
=== RUN   TestAccNetworkingVip_basic
=== PAUSE TestAccNetworkingVip_basic
=== RUN   TestAccNetworkingVip_ipv6
=== PAUSE TestAccNetworkingVip_ipv6
=== CONT  TestAccNetworkingVip_basic
=== CONT  TestAccNetworkingVip_ipv6
--- PASS: TestAccNetworkingVip_ipv6 (134.16s)
--- PASS: TestAccNetworkingVip_basic (163.07s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpc      163.189s
```

### CODE REVIEW

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/vpc' TESTARGS='-run=TestAccNetworkingVip_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpc -v -run=TestAccNetworkingVip_ -timeout 360m -parallel 4
=== RUN   TestAccNetworkingVip_basic
=== PAUSE TestAccNetworkingVip_basic
=== RUN   TestAccNetworkingVip_ipv6
=== PAUSE TestAccNetworkingVip_ipv6
=== CONT  TestAccNetworkingVip_basic
=== CONT  TestAccNetworkingVip_ipv6
--- PASS: TestAccNetworkingVip_ipv6 (111.26s)
--- PASS: TestAccNetworkingVip_basic (129.33s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpc129.393s
```
```
make testacc TEST='./huaweicloud' TESTARGS='-run=TestAccNetworkingV2VIPAssociate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run=TestAccNetworkingV2VIPAssociate_basic -timeout 360m -parallel 4
=== RUN   TestAccNetworkingV2VIPAssociate_basic
=== PAUSE TestAccNetworkingV2VIPAssociate_basic
=== CONT  TestAccNetworkingV2VIPAssociate_basic
--- PASS: TestAccNetworkingV2VIPAssociate_basic (185.32s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud      185.412s
```

**Comment by ShiChangkuo** [comment]
please remove the sentence ```This is an alternative to ...```

File: huaweicloud/services/vpc/resource_huaweicloud_networking_vip.go [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+		return common.CheckDeletedDiag(d, err, "VPC network VIP")
+	}
+
+	logp.Printf("[DEBUG] Retrieved VIP %s: %+v", d.Id(), vip)
+
+	d.SetId(vip.ID)
+
+	mErr := multierror.Append(nil,
+		d.Set("region", region),
+		d.Set("status", vip.Status),
```
Comment (ShiChangkuo) [inline huaweicloud/services/vpc/resource_huaweicloud_networking_vip.go]: **DOWN** is hard understanding， suggest to convert it to ACTIVE

File: docs/resources/networking_vip.md [hunk 2]
```diff
@@ -9,47 +9,56 @@ Manages a VIP resource within HuaweiCloud. This is an alternative to `huaweiclou
...
-data "huaweicloud_vpc_subnet" "mynet" {
-  name = "subnet-default"
+resource "huaweicloud_vpc_subnet" "test" {
+  ...
 }
 
-resource "huaweicloud_networking_vip" "myvip" {
-  network_id = data.huaweicloud_vpc_subnet.mynet.id
+resource "huaweicloud_networking_vip" "test" {
+  network_id = data.huaweicloud_vpc_subnet.test.id
```
Comment (niuzhenguo) [inline docs/resources/networking_vip.md]: you have updated to use resource instead of data source

File: docs/resources/networking_vip.md [hunk 3]
```diff
@@ -9,47 +9,56 @@ Manages a VIP resource within HuaweiCloud. This is an alternative to `huaweiclou
...
 The following arguments are supported:
 
-* `region` - (Optional, String, ForceNew) The region in which to create the vip resource.
-  If omitted, the provider-level region will be used. Changing this creates a new resource.
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the network VIP.
+  If omitted, the provider-level region will be used. Changing this will create a new network VIP resource.
 
-* `network_id` - (Required, String, ForceNew) Specifies the ID of the network to which the vip belongs.
-  Changing this creates a new resource.
+* `network_id` - (Required, String, ForceNew) Specifies the ID of the network to which the network VIP belongs.
```
Comment (niuzhenguo) [inline docs/resources/networking_vip.md]: better to make it clear that the network id is the subnet Network ID.

File: docs/resources/networking_vip.md [hunk 4]
```diff
@@ -9,47 +9,56 @@ Manages a VIP resource within HuaweiCloud. This is an alternative to `huaweiclou
...
 
-* `ip_version` - (Optional, Int, ForceNew) Specifies the IP version, either 4 (default) or 6.
-  Changing this creates a new resource.
+* `ip_address` - (Optional, String, ForceNew) Specifies the IP address desired in the subnet for this network VIP.
+  Changing this will create a new network VIP resource.
 
-* `ip_address` - (Optional, String, ForceNew) Specifies the IP address desired in the subnet for this vip.
-  Changing this creates a new resource.
-
-* `name` - (Optional, String) Specifies a unique name for the vip.
```
Comment (niuzhenguo) [inline docs/resources/networking_vip.md:37]: As we discussed, keep the `name` as is.

File: huaweicloud/services/vpc/resource_huaweicloud_networking_vip.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/vpc/resource_huaweicloud_networking_vip.go]: After communicating with relevant personnel, we learned that the status of VIP is always `DOWN`.
A function has been added for transitioning states (DOWN to ACTIVE).

File: docs/resources/networking_vip.md (hunk 2 above)
Comment (Lance52259) [reply docs/resources/networking_vip.md]: finished.

File: docs/resources/networking_vip.md (hunk 4 above)
Comment (Lance52259) [reply docs/resources/networking_vip.md:37]: The `name` has been reverted.

File: docs/resources/networking_vip.md (hunk 3 above)
Comment (Lance52259) [reply docs/resources/networking_vip.md]: yeah, the description should be clearer.

File: huaweicloud/services/vpc/resource_huaweicloud_networking_vip.go [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+		Delay:      3 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(vip.ID)
```
Comment (ShiChangkuo) [inline huaweicloud/services/vpc/resource_huaweicloud_networking_vip.go]: should set the resource ID after it was created successfully

File: huaweicloud/services/acceptance/vpc/resource_huaweicloud_networking_vip_test.go [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccNetworkingV2VIPConfig_ipv4(rName),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(resourceName, "ip_version", "4"),
+					resource.TestCheckResourceAttrSet(resourceName, "mac_address"),
+					// Deprecated parameter
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/vpc/resource_huaweicloud_networking_vip_test.go]: `name` is not a deprecated parameter.

Comment (Lance52259) [reply huaweicloud/services/acceptance/vpc/resource_huaweicloud_networking_vip_test.go]: fixed

File: huaweicloud/services/vpc/resource_huaweicloud_networking_vip.go (hunk 5 above)
Comment (Lance52259) [reply huaweicloud/services/vpc/resource_huaweicloud_networking_vip.go]: fixed.
Another place where the code for setting the ID has been removed.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/vpc' TESTARGS='-run=TestAccNetworkingVip_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpc -v -run=TestAccNetworkingVip_ -timeout 360m -parallel 4
=== RUN   TestAccNetworkingVip_basic
=== PAUSE TestAccNetworkingVip_basic
=== RUN   TestAccNetworkingVip_ipv6
=== PAUSE TestAccNetworkingVip_ipv6
=== CONT  TestAccNetworkingVip_basic
=== CONT  TestAccNetworkingVip_ipv6
--- PASS: TestAccNetworkingVip_ipv6 (118.17s)
--- PASS: TestAccNetworkingVip_basic (161.32s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpc       161.639s
```

**Pull request merged**